shell/platform/common/cpp/text_input_model.cc

```diff
@@ -113,6 +119,42 @@ bool TextInputModel::Delete() {
   return false;
 }
 
+bool TextInputModel::DeleteSurrounding(int offset, int count) {
+  auto start = selection_extent_;
+  if (offset < 0) {
+    for (int i = 0; i < -offset; i++) {
+      // If requested start is before the available text then reduce the
+      // number of characters to delete.
+      if (start == text_.begin()) {
+        count = i;
+        break;
+      }
+      start -= IsTrailingSurrogate(*(start - 1)) ? 2 : 1;
+    }
+  } else {
+    for (int i = 0; i < offset && start != text_.end(); i++)
```

Comment:
Generally we try to keep styles consistent at the file level for things that the style guide is flexible on, and this file uses braces on all `for` and `if` statements, so I'd prefer this new code to be consistent with that.

(Applies several places below as well.)